tests/test_core/test_annotation_db.py

```diff
@@ -1207,3 +1222,64 @@ def test_subset_gff3_db_source(gff_db, tmp_dir):
     # manual inspection of the original GFF3 file indicates 7 records
     # BUT the CDS records get merged into a single row
     assert len(subset) == 6
+
+
+@pytest.mark.parametrize(
+    "new_span", ([[22, 24]], [[9, 20]], [[0, 1]], [[9, 20], [29, 45], [59, 70]])
+)
+def test_gffdb_update_record(gff_db, new_span):
+    name = "CDS:B0019.1"
+    init_value = [[9, 20], [29, 45], [59, 70]]
+    combined = set(tuple(c) for c in new_span + init_value)
+    expect = [list(p) for p in sorted(combined)]
+    gff_db.update_record_spans(name=name, spans=new_span)
+    got = list(gff_db.get_records_matching(name=name))[0]
+    assert got["spans"].tolist() == expect
+
+
+def test_gffdb_update_record_empty(gff_db):
+    name = "CDS:B0019.1"
+    init_value = [[9, 20], [29, 45], [59, 70]]
+    gff_db.update_record_spans(name=name, spans=[])
+    got = list(gff_db.get_records_matching(name=name))[0]
+    assert got["spans"].tolist() == init_value
+
+
+def test_gffdb_update_absent_record(gff_db):
+    name = "qwerty"  # does not exist
+    init_value = [[9, 20], [29, 45], [59, 70]]
+    gff_db.update_record_spans(name=name, spans=init_value)
+    got = list(gff_db.get_records_matching(name=name))
+    assert not got
+
+
+@pytest.mark.parametrize("db", ("gff_db", "gb_db"))
+def test_db_close(request, db):
+    import sqlite3
+
+    db = request.getfixturevalue(db)
+    db.close()
+    with pytest.raises(sqlite3.ProgrammingError):
+        _ = list(db.get_features_matching(biotype="gene"))
+
+
+@pytest.mark.parametrize("db", ("gff_db", "gb_db"))
+@pytest.mark.parametrize(
+    "col", ("biotype", "seqid", "name", "start", "stop", "parent_id")
+)
+def test_db_make_index(request, db, col):
+    ann_db = request.getfixturevalue(db)
+    expect = {("index", col, tn) for tn in ann_db.table_names}
+    ann_db.make_indexes()
+    sql_template = f"SELECT * FROM sqlite_master WHERE type = 'index' AND tbl_name = '%s' and name = {col!r}"
```

Comment:
You might be able to ignore the codacy sql injection warning with 

`# codacy:ignore[sql-injection] - limited SQL injection exposure`

As the only exposure I can see here is to unrestricted browsing, and it's just a test

Reply:
Good point, will apply that.